### DISCUSSION TITLE
motion plot fails after eddy_quad generates an error

### DISCUSSION THREAD

**Discussion by zeffiro** [muscbridge/PyDKE#305] [Technical Support]
**Using this neurodock command line:**

docker run -it --rm -v E:/BEAP-dMRI/sub-BEAP001/ses-1/dwi:/data dmri/neurodock pydesigner \
--standard --verbose --force --rpe_pairs 1 --outp /data/derivative /data/ABCD_DMRI_0015 \ /data/ABCD_DMRI_DISTORTIONMAP_AP_0017 /data/ABCD_DMRI_DISTORTIONMAP_PA_0016

**The motion plot display fails following eddy_quad error.**
...
Command:  eddy_cpu --imain=eddy_in.nii --mask=eddy_mask.nii --acqp=eddy_config.txt --index=eddy_indices.txt --bvecs=bvecs --bvals=bvals --topup=field --repol --data_is_shelled --out=dwi_post_eddy --verbose
dwifslpreproc: [100%] Removing image padding prior to running EddyQC
Command:  eddy_quad dwi_post_eddy_unpad -idx eddy_indices.txt -par eddy_config.txt -b bvals -m eddy_mask_unpad.nii -f field_map_unpad.nii
dwifslpreproc: [WARNING] Error running automated EddyQC tool 'eddy_quad'; QC data written to "/data/derivative/metrics_qc/eddy" will be files from "eddy" only
Command:  mrconvert dwi_post_eddy.nii.gz result.mif -coord 2 0:80 -strides -1,-3,4,2 -fslgrad dwi_post_eddy.eddy_rotated_bvecs bvals
Function: shutil.copy('dwi_post_eddy_unpad.eddy_outlier_map', '/data/derivative/metrics_qc/eddy/eddy_outlier_map')
Function: shutil.copy('dwi_post_eddy_unpad.eddy_outlier_n_stdev_map', '/data/derivative/metrics_qc/eddy/eddy_outlier_n_stdev_map')
Function: shutil.copy('dwi_post_eddy_unpad.eddy_outlier_n_sqr_stdev_map', '/data/derivative/metrics_qc/eddy/eddy_outlier_n_sqr_stdev_map')
Command:  mrconvert dwi_post_eddy_unpad.eddy_outlier_free_data.nii.gz /data/derivative/metrics_qc/eddy/eddy_outlier_free_data.nii.gz -coord 2 0:80
Command:  mrconvert eddy_mask.nii /data/derivative/metrics_qc/eddy/eddy_mask.nii -coord 2 0:80
Command:  mrconvert result.mif /data/derivative/2_dwi_undistorted.mif
dwifslpreproc: Changing back to original directory (/data/derivative)
dwifslpreproc: Deleting scratch directory (/data/derivative/dwifslpreproc-tmp-RCR5A3/)
Plotting motion...
Traceback (most recent call last):
  File "/usr/local/bin/pydesigner", line 10, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/dist-packages/designer/pydesigner.py", line 674, in main
    voxel=mrinfoutil.spacing(working_path))
  File "/usr/local/lib/python3.7/dist-packages/designer/plotting/motionplot.py", line 33, in plot
    raise OSError('Input file {} does not exist'.format(input))
OSError: Input file /data/derivative/metrics_qc/eddy/eddy_restricted_movement_rms does not exist

**Any ideas?**

**Comment by zeffiro** [comment]
UPDATE:
Running the command with the --noqc switch seems to pypass the problem and allow the script to complete, but we are not getting any QC information. Is there a possible fix?

**Reply by TheJaeger** [reply]
Thanks for bringing this up @zeffiro. It seems like PyDesigner could use work on the logic that determines when to create motion plot. The only QC metic it wouldn't produce due to this is the motion plot. However, other QC metrics like SNR and IRLLS outliers should still be produced.

**Comment by zeffiro** [comment]
Suppressing the motion as an option might be a good addition. Currently, when the motion plot fails, all the intermediate files are deleted during the cleanup before exit. Maybe the temporary intermediate file deletion in Neurodock could be suppressed with a command line option. That would facilitate debugging in these cases that cause things to fail.
